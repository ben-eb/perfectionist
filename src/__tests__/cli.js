import child from 'child_process';
import path from 'path';
import fs from 'fs';
import ava from 'ava';

let read = fs.readFileSync;
let fixture = path.join(__dirname, './fixtures/nested.fixture.css');

function setup (args) {
    return new Promise(resolve => {
        process.chdir(__dirname);

        let ps = child.spawn(process.execPath, [
            path.resolve(__dirname, '../../bin/cmd.js'),
        ].concat(args));

        let out = '';
        let err = '';

        ps.stdout.on('data', buffer => (out += buffer));
        ps.stderr.on('data', buffer => (err += buffer));

        ps.on('exit', code => {
            resolve([err, out, code]);
        });
    });
}

ava('cli: defaults', t => {
    return setup([fixture]).then(([err, out, code]) => {
        t.falsy(err, 'should not error');
        t.falsy(code, 'should exit with code 0');
        t.deepEqual(out, read('./fixtures/nested.expanded.css', 'utf-8'), 'should transform the css');
    });
});

ava('cli: formatter', t => {
    return setup([fixture, '--format', 'compressed']).then(([err, out, code]) => {
        t.falsy(err, 'should not error');
        t.falsy(code, 'should exit with code 0');
        t.deepEqual(out, read('./fixtures/nested.compressed.css', 'utf-8'), 'should transform the css');
    });
});

ava('cli: sourcemaps', t => {
    return setup([fixture, '--sourcemap']).then(([err, out]) => {
        t.falsy(err);
        const hasMap = /sourceMappingURL=data:application\/json;base64/.test(out);
        t.truthy(hasMap, 'should generate a sourcemap');
    });
});
