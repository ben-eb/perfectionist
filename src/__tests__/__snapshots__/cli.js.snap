// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli: defaults 1`] = `
"h1 {
    color: red;

    & h2 {
        color: yellow;

        & h3 {
            color: green;
        }
    }
}
"
`;

exports[`cli: formatter 1`] = `
"h1 {
    color: red;

    & h2 {
        color: yellow;

        & h3 {
            color: green;
        }
    }
}
"
`;
